reactor-core/src/main/java/reactor/util/context/ContextN.java

```diff
@@ -16,20 +16,19 @@
 package reactor.util.context;
 
 import java.util.Collections;
-import java.util.HashMap;
-import java.util.List;
 import java.util.Map;
 import java.util.Map.Entry;
 import java.util.NoSuchElementException;
 import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.function.Function;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
 import reactor.util.annotation.Nullable;
 
 @SuppressWarnings("unchecked")
-final class ContextN extends HashMap<Object, Object>
+final class ContextN extends ConcurrentHashMap<Object, Object>
```

Comment:
Why do we need to extend from `ConcurrentHashMap` now?

Reply:
as stated in the commit and pr message, `HashMap` doesn't reject `null` keys nor values. `Context` is supposed to. `ConcurrentHashMap` rejects null keys and values.

Reply:
I think using `ConcurrentHashMap` just for the null checks is an overkill given that it comes with performance overhead for single-threaded usage

Reply:
then we'll have to devolve to avoiding `super.putAll` and `super.put` in favor of a loop over entries with a null check 👍

Reply:
@bsideup updated